core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,125 @@
+package ocr2keeper
+
+import (
+	"context"
+	"time"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+
+	httypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/headtracker/types"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/static"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+const (
+	customTelemChanSize = 100
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint commontypes.MonitoringEndpoint
+	headBroadcaster    httypes.HeadBroadcaster
+	headCh             chan blockKey
+	unsubscribe        func()
+	chDone             chan struct{}
+	lggr               logger.Logger
+}
+
+// NewAutomationCustomTelemetryService creates a telemetry service for new blocks and node version
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint, hb httypes.HeadBroadcaster, lggr logger.Logger) *AutomationCustomTelemetryService {
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint: me,
+		headBroadcaster:    hb,
+		headCh:             make(chan blockKey, customTelemChanSize),
+		chDone:             make(chan struct{}),
+		lggr:               lggr.Named("Automation Custom Telem"),
+	}
+}
+
+// Start starts Custom Telemetry Service, sends 1 NodeVersion message to endpoint at start and sends new BlockNumber messages
+func (e *AutomationCustomTelemetryService) Start(context.Context) error {
+	return e.StartOnce("AutomationCustomTelemetryService", func() error {
+		e.lggr.Infof("Starting: Custom Telemetry Service")
+		versionMsg := &telem.NodeVersion{
+			Timestamp:   uint64(time.Now().UTC().UnixMilli()),
+			NodeVersion: static.Version,
+		}
+		wrappedMessage := &telem.AutomationTelemWrapper{
+			Msg: &telem.AutomationTelemWrapper_NodeVersion{
+				NodeVersion: versionMsg,
+			},
+		}
+		bytes, err := proto.Marshal(wrappedMessage)
+		if err != nil {
+			e.lggr.Errorf("Error occurred while marshalling the message: %v", err)
+		}
```

Comment:
we are just logging the error and continuing but the line below send `bytes` which is potential nil is that ok/safe?